runtime/onert/core/include/exec/FunctionSequence.h

```diff
@@ -72,33 +74,76 @@ class FunctionSequence : public IFunction
     }
   }
 
-protected:
-  std::vector<std::unique_ptr<IFunction>> _functions;
-};
+public: // context for dynamic tensor
+  /**
+   * @brief
+   *
+   */
+  void doNotRunDynamicShapeInferer() { _run_dynamic_shape_inferer = false; }
 
-/**
- * @brief Function sequence used for backend that supports dynamic tensor
- *        Such backend cannot use class FunctionSequence but use this class
- */
-class FunctionSequenceForDynamicBackend : public FunctionSequence
-{
-public:
-  FunctionSequenceForDynamicBackend(const ir::OpSequence &op_seq, const ir::Operations &operations,
-                                    std::unique_ptr<DynamicInferer> dyn_shape_inferer,
-                                    backend::IDynamicTensorManager *dyn_tensor_manager)
-      : _op_seq(op_seq), _operations_ctx(operations),
-        _dyn_shape_inferer(std::move(dyn_shape_inferer)), _dyn_tensor_manager(dyn_tensor_manager)
-  { /* empty */
+  /**
+   * @brief Prepare to run FunctionSequence which "might" handle dynamic tensor
+   * @note  Calling this does not mean that run() will handle dynamic tensor.
+   *        hasDynamicTensor(true) will make run() will handle dynamic tensor.
+   */
+  void dynamicTensorContext(const ir::OpSequence *op_seq, const ir::Operations *operations,
```

Comment:
- It should be named `dynamic_tensor_context` or `setDynamicTensorContext` according to our coding convention
- What about packing 5 argument as a struct?
- What about moving this to the constructor? Calling this function seems like a must